### PR TITLE
check if the set is empty , and delete the reduntant type convert.

### PR DESCRIPTION
**Pull request by tanshanshan**
**What this PR does / why we need it**:

check if the set is empty , and delete the reduntant type convert.

Thanks.

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by lavalamp** [comment]
@k8s-bot ok to test

/lgtm

Sorry for delay in reviewing, still catching up from being OOO.

**Comment by tanshanshan** [comment]
I have rebased it , PTAL.

File: staging/src/k8s.io/apimachinery/pkg/labels/selector.go [hunk 1]
```diff
@@ -793,7 +793,7 @@ func validateLabelValue(v string) error {
 // SelectorFromSet returns a Selector which will match exactly the given Set. A
 // nil and empty Sets are considered equivalent to Everything().
 func SelectorFromSet(ls Set) Selector {
-	if ls == nil {
+	if ls == nil || len(ls) == 0 {
```
Comment (lavalamp) [inline staging/src/k8s.io/apimachinery/pkg/labels/selector.go]: Just FYI, it's safe to call len on nil maps so you could have just replaced the == nil with the clause you added.

**Comment by lavalamp** [comment]
@k8s-bot gce etcd3 e2e test this

**Pull request merged**